### DISCUSSION TITLE
Filamentphp, best practice for simple API service for public consumption

### DISCUSSION THREAD

**Discussion by LukaszJaro** [filamentphp/filament#12969] [Help]
### Package

Other

### Package Version

v3.0.0

### How can we help you?

I just need to create a route that lists data in JSON format from filamentphp (GET method only) and allow the route to be publicly accessible. Is their any documentation on this or best practice to follow? Filamentphp is not using controllers so would I need to create a controller?

**Comment by leandrocfe** [comment]
Something like this?
https://filamentphp.com/plugins/rupadana-api-service

**Reply by LukaszJaro** [reply]
Thanks, I'm aware of that plugin, I just wanted something very simple with no other dependency. I used this from documentation and it's working:

1. php artisan make:resource YourResource (don't use filament resource command)
2. Then add the api config which was removed in Laravel 11: php artisan install:api

https://laravel.com/docs/11.x/eloquent-resources
https://laravel.com/docs/11.x/routing#api-routes
